docs/Advanced-Usage/Heterogeneous-SoCs.rst

```diff
@@ -0,0 +1,127 @@
+
+.. code-block:: scala
+
+    class DualBoomAndOneRocketConfig extends Config(
+      new WithNormalBoomRocketTop ++
+      new WithBootROM ++
+      new boom.system.WithRenumberHarts ++
+      new boom.common.WithRVC ++
+      new boom.common.DefaultBoomConfig ++
+      new boom.system.WithNBoomCores(2) ++
+      new freechips.rocketchip.subsystem.WithoutTLMonitors ++
+      new freechips.rocketchip.subsystem.WithNBigCores(1) ++
+      new freechips.rocketchip.system.BaseConfig)
+
+In this example, the ``WithN...Cores(X)`` mixins set up particular parameters for the duplicated core.
+For example, the Rocket tiles will be "BigCores" while BOOM has some generic parameters set in its mixin.
+In BOOMs case, we override these base core parameters to create multiple "DefaultBoom"s with calling ``DefaultBoomConfig`` after the multi-core instantiation.
+This mixin applies to all the BOOM cores in the system and sets up the parameters for each.
+
+Great! Now you have a heterogeneous setup with BOOMs and Rockets.
+The final thing you need to make this system work is to renumber the ``hartId``'s of the cores so that each core has a unique ``hartId``.
+This is done with ``WithRenumberHarts`` (which can label the Rocket cores first or the BOOM cores first).
+The reason this is needed is because by default the ``WithN...Cores(X)`` mixin assumes that there are only BOOM or only Rocket cores in the system.
+Thus, without the ``WithRenumberHarts`` mixin, each set of cores is labeled starting from zero causing multiple cores to be assigned the same ``hartId``.
+
+Another option to create a multi heterogeneous core system is to override the parameters yourself so you can specify the core parameters per core.
+The mixin to add to your system would look something like the following.
+
+.. code-block:: scala
+
+    // create 6 cores (4 boom and 2 rocket)
+    class WithHeterCoresSetup extends Config((site, here, up) => {
+      case BoomTilesKey => {
+        val boomTile0 = BoomTileParams(...) // params for boom core 0
+        val boomTile1 = BoomTileParams(...) // params for boom core 1
+        val boomTile2 = BoomTileParams(...) // params for boom core 2
+        val boomTile3 = BoomTileParams(...) // params for boom core 3
+        boomTile0 ++ boomTile1 ++ boomTile2 ++ boomTile3
+      }
+
+      case RocketTilesKey => {
+        val rocketTile0 = RocketTileParams(...) // params for rocket core 0
+        val rocketTile1 = RocketTileParams(...) // params for rocket core 1
+        rocketTile0 ++ rocketTile1
+      }
+    })
+
+Then you could use this new mixin like the following.
+
+.. code-block:: scala
+
+    class SixCoreConfig extends Config(
+      new WithNormalBoomRocketTop ++
+      new WithBootROM ++
+      new WithHeterCoresSetup ++
+      new freechips.rocketchip.system.BaseConfig)
+
+Note, in this setup you most likely don't need the ``WithRenumberHarts`` mixin since you have to assign the ``hartId`` of each tile in the tile parameters yourself.
```

Comment:
"most likely"?

Reply:
So.... if they take the default it will be broken because all cores with get `hartId == 0`. So they will have to set the `hartId`. I will make this clearer.